### ISSUE TITLE
How to enable WebGL stage debugging?

### ISSUE THREAD

**Issue by VepretskiyD** [google/marzipano#292]
On some devices and browsers, the viewer doesn't show anything (blank screen), freezes and stops responding.
https://github.com/google/marzipano/blob/265f1c8de111129e3237b8355be3f4c06514ac6e/src/support/WebGl.js#L20
checkWebGlSupported() check passes but for some reasons rendering fails.
I noticed debug mode in WebGl.js. It gets enabled if `typeof MARZIPANODEBUG !== 'undefined' && MARZIPANODEBUG.webGl` and `typeof WebGLDebugUtils !== 'undefined'`. I guess it was added for development of the viewer itself and not for public use.

https://github.com/google/marzipano/blob/265f1c8de111129e3237b8355be3f4c06514ac6e/src/stages/WebGl.js#L29
https://github.com/google/marzipano/blob/265f1c8de111129e3237b8355be3f4c06514ac6e/src/stages/WebGl.js#L62

I need to listen to WebGL errors and at least throw them to the browser's console where they will be caught by Sentry. Ideally, I want to show some message in UI that this device and a browser experience issues with WebGL.
I've found this tool very usefull https://github.com/KhronosGroup/WebGLDeveloperTools/blob/master/src/debug/webgl-debug.js
I was able to get webGLcontext and wrap it with WebGLDebugUtils:

```
      let gl = viewer.stage().webGlContext();
      function throwOnGLError(err, funcName, args) {
        throw `${WebGLDebugUtils.glEnumToString(err)} was caused by call to: ${funcName}`;
      }
      function logGLCall(functionName, args) {
        console.log(`gl.${functionName}(${WebGLDebugUtils.glFunctionArgsToString(functionName, args)})`);
      }
      function validateNoneOfTheArgsAreUndefined(functionName, args) {
        for (let ii = 0; ii < args.length; ++ii) {
          if (args[ii] === undefined) {
            console.error(`undefined passed to gl.${functionName}(${WebGLDebugUtils.glFunctionArgsToString(functionName, args)})`);
          }
        }
      }
      function logAndValidate(functionName, args) {
        logGLCall(functionName, args);
        validateNoneOfTheArgsAreUndefined(functionName, args);
      }
      gl = WebGLDebugUtils.makeDebugContext(gl, throwOnGLError, logAndValidate);
```
Unfortunatelly it doesn't work as intended. I guess, I wrap the webGlContext too late, after the viewer was instantiated.
Is it possible to listen to WebGL errors without changing the source code?

**Comment by tjgq** [comment]
Yes, you're probably wrapping the context too late. The `MARZIPANODEBUG` bits were used during development, but they're not particularly useful in their current form.

Can you try out https://github.com/google/marzipano/pull/293? It makes it possible to pass an arbitrary function to wrap the rendering context into the Stage or Viewer constructor, e.g.

```
function myWrapper(gl) {
    return WebGLDebugUtils.makeDebugContext(gl, throwOnGLError, logAndValidate);
}

const viewer = new Marzipano.Viewer({ stage: { wrapContext: myWrapper }});
```

If that PR works for you, I can merge it into the next release.

**Comment by tjgq** [comment]
The PR mentioned above was merged in 660c1d6.
